cpp/include/cuvs/neighbors/hnsw.hpp

```diff
@@ -303,7 +303,7 @@ void search(raft::resources const& res,
 void search(raft::resources const& res,
             const search_params& params,
             const index<int8_t>& idx,
-            raft::host_matrix_view<const int, int64_t, raft::row_major> queries,
+            raft::host_matrix_view<const int8_t, int64_t, raft::row_major> queries,
             raft::host_matrix_view<uint64_t, int64_t, raft::row_major> neighbors,
             raft::host_matrix_view<float, int64_t, raft::row_major> distances);
```

Comment:
We are ultimatley going to need to support filtered search here. Can you create an issue for this and reference it in a todo in this file? Beginning of the file will be fine or just before all the search functions.